src/Datadog.Trace/AspNetCore/DatadogLoggingScope.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="DatadogLoggingScope.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Datadog.Trace.AspNetCore
+{
+    // Implementation based on AspNetCore Hosting implementation: https://github.com/dotnet/aspnetcore/blob/1a2b3260c6161ae9b7f639de228a6eb0488a1075/src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs#L114
+    internal class DatadogLoggingScope : IReadOnlyList<KeyValuePair<string, object>>
```

Comment:
Wonder if we should make this public, so customers can use it with manual instrumentation?

Reply:
Oh, is this for non-aspnetcore use cases? I was thinking we would start by only exposing an ApplicationBuilder extension method that the user can call, and this could be an implementation detail.

Reply:
Yeah, I was thinking non-aspnetcore use cases. The `ApplicationBuilder` makes sense, but then we would have to take dependencies on the aspnetcore packages, which I wasn't sure if we'd want to? We can always make this public later if we want to, so probably makes sense to keep it internal for now.

Reply:
>but then we would have to take dependencies on the aspnetcore packages

In a new [`Datadog.Trace.AspNetCore`](https://www.nuget.org/packages/Datadog.Trace.AspNetCore/) package! 😅